### PR TITLE
init the gapi once so that it can chrome will not block the login popup

### PR DESCRIPTION
**Pull request by darrenchiu**
for #4 , the only thing that I am not sure is that I can only access gapi by `window.gapi`, not sure if that the correct approach.

didn't do the dist build as well to save some setup time, thanks

### CODE REVIEW

**Comment by kennetpostigo** [comment]
@darrenchiu By any chance were you able to test this on the example CRA site?

**Comment by D-Andreev** [comment]
Hey @darrenchiu , thanks for making this PR. If this fix works, can you merge it and publish on npm? I'm currently using `react-google-login-component` in a project and it would be great to have this in. Thanks!

**Comment by kennetpostigo** [comment]
@D-Andreev do you think you can see if this change on his fork works on the example? If it does I will merge and publish immediately

File: src/GoogleLogin.js [hunk 1]
```diff
@@ -1,54 +1,64 @@
...
   componentDidMount () {
-    (function(d, s, id){
-     var js, gs = d.getElementsByTagName(s)[0];
-     if (d.getElementById(id)) {return;}
-     js = d.createElement(s); js.id = id;
-     js.src = 'https://apis.google.com/js/platform.js'
-     gs.parentNode.insertBefore(js, gs);
-   }(document, 'script', 'google-platform'));
+    window.googlePlatformReadyCb = () => {
+      console.info('gapi ready.');
```
Comment (D-Andreev) [inline src/GoogleLogin.js:13]: This should be removed.

**Review by D-Andreev — APPROVED** [review]
Works well, LGTM 👍

File: src/GoogleLogin.js (hunk 1 above)
Comment (kennetpostigo) [reply src/GoogleLogin.js:13]: Do you want to build on this PR and then create another PR to be merged @D-Andreev ? If you do I will gladly review, merge, and publish. Also thanks for helping out @D-Andreev much appreciated and welcome 😀

Comment (D-Andreev) [reply src/GoogleLogin.js:13]: Yeah sure I can  do that. Looking at https://developers.google.com/api-client-library/javascript/reference/referencedocs some other props can be added. I'll look into it in more detail the next couple of days and prepare a PR.

File: src/GoogleLogin.js [hunk 2]
```diff
@@ -1,54 +1,64 @@
...
+      js.src = 'https://apis.google.com/js/platform.js?onload=googlePlatformReadyCb';
+      gs.parentNode.insertBefore(js, gs);
+    }(document, 'script', 'google-platform'));
+
+    var socialId = this.props.socialId,
+      responseHandler = this.props.responseHandler,
+      scope = this.props.scope;
   }
 
   checkLoginState (response) {
```
Comment (D-Andreev) [inline src/GoogleLogin.js:37]: Is this actually used anywhere?

Comment (kennetpostigo) [reply src/GoogleLogin.js:37]: Yea, this would run the code you have in your handler function

File: src/GoogleLogin.js (hunk 1 above)
Comment (kennetpostigo) [reply src/GoogleLogin.js:13]: Thanks @D-Andreev !

Comment (darrenchiu) [reply src/GoogleLogin.js:13]: sorry @kennetpostigo was a bit busy and didn't keep track of the thread and thanks @D-Andreev for checking my changes

**Comment by juank11memphis** [comment]
@kennetpostigo any idea on when this PR is gonna be merged or otherwise, when the popup block problem will be fixed? 
Thanks.

**Comment by kennetpostigo** [comment]
This has been fixed in #15

**Pull request closed without merging**